sdk/monitor/azure-monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,214 @@
+    "message": "A semantic error occurred.",
+    "innererror": {
+        "code": "SEM0100",
+        "message": "'take' operator: Failed to resolve table or column expression named 'Appquests'"
+    }
+}
+```
+
+The error message in the innererror may include the where the Kusto query has an error. You may also refer to the [Kusto Query Language](https://docs.microsoft.com/azure/data-explorer/kusto/query) reference docs to learn more about querying logs using KQL.
+
+### Troubleshooting empty log query results
+
+If your Kusto query returns empty no logs, please validate the following:
+
+- You have the right workspace ID
+- You are setting the correct time interval for the query. Try expanding the time interval for your query to see if that
+  returns any results.
+- If your Kusto query also has a time interval, the query is evaluated for the intersection of the time interval in the
+  query string and the time interval set in the `timespan` param provided the query API. The intersection of
+  these time intervals may not have any logs. To avoid any confusion, it's recommended to remove any time interval in
+  the Kusto query string and use `timespan` explicitly. Please note that the `timespan` param can be a timedelta,
+  a timedelta and a start datetime, or a start datetime/end datetime.
+
+### Troubleshooting server timeouts when executing logs query request
+
+Some complex Kusto queries can take a long time to complete and such queries are aborted by the
+service if they run for more than 3 minutes. For such scenarios, the query APIs on `LogsQueryClient`, provide options to
+configure the timeout on the server. The server timeout can be extended up to 10 minutes.
+
+You may see an error as follows:
+
+```text
+Code: GatewayTimeout
+Message: Gateway timeout
+Inner error: {
+    "code": "GatewayTimeout",
+    "message": "Unable to unzip response"
+}
+```
+
+The following code shows a sample on how to set the server timeout to 10 minutes. Note that by setting this server
+timeout, the Azure Monitor Query library will automatically also extend the client timeout to wait for 10 minutes for
+the server to respond. You don't need to configure your HTTP client to extend the response timeout as shown in the
+previous section.
+
+```python
+from azure.monitor.query import LogsQueryClient
+from azure.identity import DefaultAzureCredential
+
+credential = DefaultAzureCredential()
+
+client = LogsQueryClient(credential)
+
+client.query_workspace(
+    "{workspaceId}",
+    "{kusto-query-string}",
+    timespan="{timespan}",
+    server_timeout=600)
+```
+
+### Troubleshooting partially successful logs query requests
+
+By default, if the execution of a Kusto query resulted in a partially successful response, the Azure Monitor Query
+client library will throw an exception to indicate to the user that the query was not fully successful. The data and
+the error can be accessed using the `partial_data` and `partial_error` fields.
```

Comment:
I don't think this is accurate in Python's case as it won't throw an exception. Instead, the partial results are available in `response.partial_*`. There's no switch in Python to turn on/off this behavior right?

Reply:
Yes I agree. Both Python and JS don't throw an exception